src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1105,6 +1105,20 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         }
     }
 
+    void visit_StringItem(const ASR::StringItem_t& x) {
+        if (x.m_value) {
+            this->visit_expr_wrapper(x.m_value, true);
+            return;
+        }
+        this->visit_expr_wrapper(x.m_idx, true);
+        llvm::Value *idx = tmp;
+        this->visit_expr_wrapper(x.m_arg, true);
+        llvm::Value *str = tmp;
+        llvm::AllocaInst *parg = builder->CreateAlloca(character_type, nullptr);
+        builder->CreateStore(tmp, parg);
```

Comment:
What do these two lines do? They seem to move `tmp` to `parg`, but `parg` is never used?

Reply:
I used `AllocaInst` to create a pointer to allocate memory for the `character_type` on the stack. And we use `CreateStore` to take that to `tmp`. The same thing is done in b07a4b31e998ca4f8a6081c221cd985367df9e1d.

Reply:
It's the other way round: it stores the content of `tmp` into `parg`. But you don't use `parg`, so it's not needed. in https://github.com/lcompilers/lpython/commit/b07a4b31e998ca4f8a6081c221cd985367df9e1d the `parg` is passed as an argument to `lfortran_str_len`.

Reply:
I think this `StringItem` is never used. I think the bug is that you have to replace `str` with `parg`.

I think you have to hook up `StringItem` in AST->ASR.